File: src/BreakfastProvider.Api/Models/Requests/IngredientWasteRequest.cs
namespace BreakfastProvider.Api.Models.Requests;

public record IngredientWasteRequest
{
    public string? IngredientName { get; init; }
    public decimal QuantityWasted { get; init; }
    public string? Unit { get; init; }
    public string? RecipeName { get; init; }
    public string? Reason { get; init; }
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/request/IngredientWasteRequest.java
package io.lemonlion.breakfast.model.request;

import java.math.BigDecimal;

/** Twin of C# {@code IngredientWasteRequest}. */
public record IngredientWasteRequest(
        String ingredientName, BigDecimal quantityWasted, String unit, String recipeName, String reason) {

    public IngredientWasteRequest {
        if (quantityWasted == null) {
            quantityWasted = BigDecimal.ZERO;
        }
    }
}
